insights/specs/datasources/yum_updates.py

```diff
@@ -151,38 +161,44 @@ def load(self):
             pass
 
         try:
-            self.packages = self.base.pkgSack.returnPackages()
+            pkgs = self.base.pkgSack.returnPackages()
+            for pkg in pkgs:
+                naevr = (pkg.name, pkg.arch, pkg.epoch, pkg.version, pkg.release,)
+                self.naevr_to_repo.setdefault(naevr, set()).add(pkg.repoid)
         except yum.Errors.RepoError:
             # RepoError is raised when cache is empty
             pass
-        self.repos = self.base.repos.repos
-        self._build_updict()
-
-    def _build_updict(self):
-        self.updict = {}
-        for pkg in self.packages:
-            self.updict.setdefault(pkg.na, []).append(pkg)
 
     def installed_packages(self):
         return self.base.rpmdb.returnPackages()
 
     def updates(self, pkg):
-        nevra = pkg.nevra
-        updates_list = []
-        for upg in self.updict.get(pkg.na, []):
-            if upg.verGT(pkg):
-                updates_list.append(upg)
-        return nevra, updates_list
+        updates = self.base.upinfo.get_applicable_notices((pkg.name, pkg.arch, pkg.epoch, pkg.version, pkg.release,))
+        updates_list = set()
+        # updates are returnes in format [((name, arch, epoch, version, release), UpdateNotice)]
+        for naevr, update_notice in updates:
+            for repoid in self.naevr_to_repo[naevr]:
+                # pkg_tup is composed from ((n,a,e,v,r), repoid)
+                pkg_tup = (naevr, repoid,)
+                self.pkg_tup_to_advisories.setdefault(pkg_tup, set()).add(update_notice.get_metadata().get("update_id"))
```

Comment:
can it happen that if there is no errata, then the `update_id` exists but contains something like empty string? I just want to make sure we always add `None` to set